### PR TITLE
Xaiki/t15164 eos in containers

### PR DESCRIPTION
**Pull request by xaiki**
https://phabricator.endlessm.com/T15164

### CODE REVIEW

**Comment by cosimoc** [comment]
@xaiki Thanks! This looks reasonable to me, but are you sure that this will not result in those services being actually enabled in every image? I don't have a problem with us shipping them by default, but they should probably be disabled and not running until one enables them with `systemctl`.

@dbnicholson @jprvita Do you want to take a look at this too?

**Comment by xaiki** [comment]
they are disabled in f1733cf450e4d4d5ee308e9c87f17b458d8a12b7 by @dbnicholson

**Comment by dbnicholson** [comment]
That disables the preset, which is only run at first boot time. Looking again, we also run the preset code in the ostree builder so we don't get surprises on ostree updates. However, I'd still disable these services by default in the build so that converted users don't suddenly have 2 competing networking services.

The systemd build statically enables them during install. See the following:

https://github.com/endlessm/systemd/blob/master/Makefile.am#L5336
https://github.com/endlessm/systemd/blob/master/Makefile.am#L5585

These will create symlinks in /etc/systemd/system/multi-user.target.wants and similar, which correspond to them being enabled by default when the package is installed or updated.

So, I think you have 2 options.

1. Delete/comment those parts out in Makefile.am
2. Remove those symlinks in debian/rules in `override_dh_install`

If you add back the `save_is_enabled` check for them in debian/systemd.preinst, then you will support someone manually enabling networkd and then upgrading systemd later.

File: debian/control [hunk 1]
```diff
@@ -31,6 +31,7 @@ Build-Depends: debhelper (>= 9.20141010),
                libselinux1-dev (>= 2.1.9),
                libacl1-dev,
                liblzma-dev,
+               liblz4-dev (>= 0.0~r125),
```
Comment (dbnicholson) [inline debian/control]: Why is this being added here? It doesn't seem to have anything to do with networkd.

Comment (xaiki) [inline debian/control]: a test was failing without it.

Comment (xaiki) [inline debian/control]: that said looking into the source it's really only `test-compress` that calls it and it shouldn't be called without lz4.

it was introduced in debian in https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=832010 I guess we can remove it.. let me retry a build without it.

Comment (xaiki) [inline debian/control]: it was actually fixed by b110c04, removed it.

**Comment by cosimoc** [comment]
@xaiki Is this ready for another round?

**Comment by jprvita** [comment]
@xaiki Please fix the commit messages: wrap the body lines at col 72 and try to keep the summary at max 50 chars, with 72 as an absolute max. Right now your commit messages do not fit my fullscreen terminal with 177 cols. Also, systemd upstream does not use "Signed-off-by", and from what I can tell neither Debian, so lets follow them. The messages also need some TLC, as some have editing errors like "they will stay quiet until we  them."

Did you check if there are any symlinks pointing to `/lib/systemd/system/{systemd-networkd*,systemd-resolved.service}` after the package is first installed? I imagine you can test that with a debootstrap 
environment. I see a5186e47d0 have added:
```
# remove symlinks enabling default-on services
rm -rf debian/install/*/etc/systemd/system/*.target.wants/
```
But I'm not entirely sure if these links are created in `/etc` or `/usr` during `make install`. (I thought it was in `/usr`, but I always get confused with autotools variables).

Besides what I pointed, this looks good. Thanks for the work there!

**Comment by dbnicholson** [comment]
I didn't realize that the debian rules already stripped out the wants symlinks in /etc. In that case, you can ignore most of my comments above. systemd does create some static enabling in /lib, but it looks like in the case of networkd and resolved that it's done in /etc.

@xaiki does the build now pass all the tests after these changes? Did you check that networkd and resolved are not enabled on a fresh build like @jprvita suggested?

**Comment by xaiki** [comment]
i'm waiting for that build to finish,... what i built locally didn't install cleanly in my VM because of deps issues

**Comment by xaiki** [comment]
this now passes build thanks to @dbnicholson i'd like to get this in now, so i can resume T15164, but we should note that now can propbably re-enable some tests that were failing because of /sys not being mounted in OBS.

i tested the packages that are in my branch in a VM and i confirm networkd stays disabled.

**Pull request merged**